### PR TITLE
Fix #21 plus docs.

### LINKED ISSUE #21: Crash with Guitar Hero I songs
Hi, I've just installed FoFix git, with new configuration folder, but when trying to open a Guitar Hero I song I'm getting a crash due to:

```
File "/home/marco/Giochi/FoFix/src/views/Dialogs.py", line 1274, in keyPressed
    self.players[i].difficulty = self.info.partDifficulties[self.players[i].part.id][self.selected[i]]
IndexError: list index out of range
```

After some debugging, it seems that self.info.partDifficulties[self.players[i].part.id] is an empty list here.

All works using 3.121 provided by ubuntu repositories instead.

### PR DESCRIPTION
**Pull request by wrzwicky**
Fix #21; fix a couple other bugs; add docs.

This is a rebase of #86 off the current fofix/master, plus divided into more commits.

### CODE REVIEW

File: fofix/game/song/song.py [hunk 1]
```diff
@@ -2295,28 +2298,27 @@ def tempo(self, value):
...
 
-        if self.logSections == 1:
-            log.debug(tempText + tempText2)
-
-        self.guitarSoloIndex = 0
-        self.guitarSoloActive = False
+        if self.partnumber:
+            if text in parts[VOCAL_PART].trackName and parts[VOCAL_PART] not in self.song.parts:
+                self.useVocalTrack = False
+
```
Comment (Linkid) [inline fofix/game/song/song.py]: Remove trailing spaces.

File: fofix/game/song/song.py [hunk 2]
```diff
@@ -2718,7 +2717,10 @@ def reset(self):
         self.scores = []
 
 class MidiPartsDiffReader(midi.MidiOutStream):
-
+    """ Partially parse a MIDI file for the parts & difficulty.
+    To fully load the file, see MidiReader above.
+    """
+
```
Comment (Linkid) [inline fofix/game/song/song.py]: Remove trailing spaces.

File: fofix/game/song/song.py [hunk 3]
```diff
@@ -2800,6 +2803,8 @@ def addPart(self):
         self.nextPart = None
 
     def note_on(self, channel, note, velocity):
+        """ Note found, register its difficulty. """
+
```
Comment (Linkid) [inline fofix/game/song/song.py]: Remove trailing spaces.

File: fofix/game/song/song.py [hunk 4]
```diff
@@ -2808,21 +2813,22 @@ def note_on(self, channel, note, velocity):
...
-                diff = difficulties[track]
-                if diff not in self.difficulties[self.currentPart]:
-                    self.difficulties[self.currentPart].append(diff)
+                diffObj = difficulties[diff]
+                if diffObj not in self.difficulties[self.currentPart]:
+                    self.difficulties[self.currentPart].append(diffObj)
         except KeyError:
             pass
-
+
```
Comment (Linkid) [inline fofix/game/song/song.py]: Remove trailing spaces.

File: fofix/game/song/songconstants.py [hunk 5]
```diff
@@ -34,7 +34,12 @@
 
 
 
-# translating / marking the common MIDI notes:
+##
```
Comment (Linkid) [inline fofix/game/song/songconstants.py]: Avoid doubling `#` characters

**Comment by Linkid** [comment]
I read the code but did'nt test it yet. I'll do it as soon as possible.

**Comment by Linkid** [comment]
Thank you again for your PR. I tested it and it is good :).

Please, commit requested changes before I merge it.

File: fofix/core/Data.py [hunk 6]
```diff
@@ -421,16 +421,29 @@ def getImgDrawing(self, fileName, openImage=True):
...
-                    return True
+                try:
+                    # glob parses [] but those are legal chars on Windows, so we must escape them.
+                    # it must be done like this so replacements are not mangled by other replacements.
+                    replacements = {
+                        "[": "[[]",
+                        "]": "[]]"
+                    }
+                    fileName1 = "".join([replacements.get(c,c) for c in fileName1])
+
```
[the remark this answers is not in the document]
Comment (wrzwicky) [reply fofix/core/Data.py]: Arguable, but you asked, so 'tis done.  See:
https://stackoverflow.com/questions/2727988/python-indentation-in-empty-lines

**Comment by wrzwicky** [comment]
Note that I used autopep8 (line length 100) to clean up the blank lines, so there are **MANY** more changes than you might expect, but it's all white space and line wrapping. I hope you don't mind.

**Comment by wrzwicky** [comment]
Second batch of commit is just to fix the commit message in the first batch. Sorry for the duplicate, but Git makes this extremely hard.

**Comment by Linkid** [comment]
Ok, merged.
You can now remove your branch.

**Pull request merged**